packages/salesforcedx-vscode-core/src/util/configUtil.ts

```diff
@@ -6,79 +6,110 @@
-        return undefined;
-      }
-    }
-    return undefined;
+  /*
+   * The User-configured API version is set by the user, and is used to
+   * override the API version that is otherwise gotten from the authenticated
+   * Org in some cases, such as when deploying metadata.
+   */
+  public static async getUserConfiguredApiVersion(): Promise<
+    string | undefined
+  > {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const apiVersion = configAggregator.getPropertyValue(
+      OrgConfigProperties.ORG_API_VERSION
+    );
+    return apiVersion ? String(apiVersion) : undefined;
   }
 
-  public static async getUserConfiguredApiVersion() {
-    const apiVersion = await ConfigUtil.getConfigValue('apiVersion');
-    return apiVersion ? String(apiVersion) : undefined;
+  public static async getDefaultUsernameOrAlias(): Promise<string | undefined> {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const defaultUsernameOrAlias = configAggregator.getPropertyValue(
+      OrgConfigProperties.TARGET_ORG
+    );
+    return defaultUsernameOrAlias ? String(defaultUsernameOrAlias) : undefined;
+  }
+
+  public static async isGlobalDefaultUsername(): Promise<boolean> {
+    const configSource: ConfigSource = await ConfigUtil.getConfigSource(
+      OrgConfigProperties.TARGET_ORG
+    );
+    return configSource === ConfigSource.Global;
+  }
+
+  /*
+   * Currently, the docs tell Users to manually create this entry in the .sfdx
+   * configuration file.  For that reason, getTemplatesDirectory uses the
+   * SfdxConfigAggregator specifically.
+   */
+  public static async getTemplatesDirectory(): Promise<string | undefined> {
+    const sfdxConfigAggregator = await ConfigAggregatorProvider.getInstance().getSfdxConfigAggregator();
+    const templatesDirectory = sfdxConfigAggregator.getPropertyValue(
+      SfdxPropertyKeys.CUSTOM_ORG_METADATA_TEMPLATES
+    );
+    return templatesDirectory ? String(templatesDirectory) : undefined;
+  }
+
+  public static async isTelemetryDisabled(): Promise<boolean> {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const isTelemetryDisabled = configAggregator.getPropertyValue(
+      SfConfigProperties.DISABLE_TELEMETRY
+    );
+    return isTelemetryDisabled === 'true';
+  }
+
+  public static async getDefaultDevHubUsernameOrAlias(): Promise<
+    string | undefined
+  > {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const defaultDevHubUserName = configAggregator.getPropertyValue(
+      OrgConfigProperties.TARGET_DEV_HUB
+    );
+    return defaultDevHubUserName ? String(defaultDevHubUserName) : undefined;
+  }
+
+  public static async getGlobalDefaultDevHubUsernameOrAlias(): Promise<
+    string | undefined
+  > {
+    const globalConfig = await Config.create({ isGlobal: true });
```

Comment:
Was able to figure out how to use Config here to get the global value.  This simplifies things in the ConfigAggregatorProvider and its tests and feels like a more solid way to get the global value than trying to ensure the process is not in a project dir before instantiating the ConfigAggregator..